src/mlpack/methods/decision_tree/all_categorical_split_impl.hpp

```diff
@@ -16,17 +16,18 @@ namespace mlpack {
 namespace tree {
 
 template<typename FitnessFunction>
-template<bool UseWeights, typename VecType, typename WeightVecType>
+template<bool UseWeights, typename VecType, typename ElemType, typename WeightVecType>
 double AllCategoricalSplit<FitnessFunction>::SplitIfBetter(
     const double bestGain,
     const VecType& data,
     const size_t numCategories,
-    const arma::Row<size_t>& labels,
+    const arma::Row<ElemType>& labels,
```

Comment:
Alternately, we could use `LabelsType` or `ResponsesType` as a typedef for the labels here, instead of requiring an `arma::Row<>`.

Reply:
Just for clarification, when you say "typedef" you mean instead of `arma::Row<ElemType>`, I should use another template `LabelsType`. Or did you mean something else?

Reply:
Yes, sorry, I'm not sure why I used the word "typedef" there... I think "type" was what I meant. :)  And yeah, if you use a template `LabelsType`, it can be an Armadillo object holding whatever underlying type---including subviews.  That could help prevent copies in some situations.

Reply:
Okay. I will try to do that. 👍